### PR TITLE
BEE-41625 - UpdateApp procedure should deploy an app without a forced…

### PR DESCRIPTION
**Pull request by cmarunachalam**
BEE-41625 - UpdateApp procedure should deploy an app without forced restart of the application.
BEE-46334 - Fixed an issue *UpdateApp* procedure fails for Delete operation.

### CODE REVIEW

File: help/changelog.yaml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+3.1.0:
+  - Fixed an issue that prevented the *UpdateApp* procedure running while using *Delete* operation.
```
Comment (jsmith-cb) [inline help/changelog.yaml:2]:
```suggestion
  - Fixed an issue that prevented the *UpdateApp* procedure from running while using a *Delete* operation.
```

File: help/changelog.yaml [hunk 2]
```diff
@@ -1,3 +1,6 @@
...
+  - Provided an option in *UpdateApp* procedure to update an app without forced restart of the application/cluster.
```
Comment (jsmith-cb) [inline help/changelog.yaml:3]:
```suggestion
  - Added option in the *UpdateApp* procedure to update applications without forcing application/cluster to restart.
```

File: src/main/resources/project/parameterForms/UpdateApp.xml [hunk 3]
```diff
@@ -103,6 +103,16 @@
         <documentation>Name of the server on which application is deployed.</documentation>
         <required>0</required>
     </formElement>
+    <formElement>
+        <type>checkbox</type>
+        <label>Restart the application/cluster:</label>
+        <property>restart</property>
+        <documentation>If select, forcing to restart the application/cluster.</documentation>
```
Comment (jsmith-cb) [inline src/main/resources/project/parameterForms/UpdateApp.xml:110]:
```suggestion
        <documentation>If selected, the application/cluster is restarted to apply updates.</documentation>
```

File: src/main/resources/project/project.xml [hunk 4]
```diff
@@ -10857,6 +10882,13 @@
                 <required>0</required>
                 <type>entry</type>
             </formalParameter>
+            <formalParameter>
+                <formalParameterName>restart</formalParameterName>
+                <defaultValue />
+                <description>If select, forcing to restart the application/cluster.</description>
```
Comment (jsmith-cb) [inline src/main/resources/project/project.xml:10888]:
```suggestion
                <description>If selected, the application/cluster is restarted to apply updates.</description>
```

**Review by jsmith-cb — APPROVED** [review]
Docs-only approval

**Review by eserratocb — APPROVED** [review]

**Pull request merged**